### PR TITLE
Refs #26780 -- Made prefetch_related() don't use window expressions for sliced queries if not supported.

### PR DESCRIPTION
**Pull request by felixxm**
See https://github.com/django/django/commit/242499f2dc2bf24a9a5c855690a2e13d3303581a#r82451397

### CODE REVIEW

File: django/db/models/fields/related_descriptors.py [hunk 1]
```diff
@@ -85,13 +85,11 @@ def __set__(self, instance, value):
 
 def _filter_prefetch_queryset(queryset, field_name, instances):
     predicate = Q(**{f"{field_name}__in": instances})
-    if queryset.query.is_sliced:
+    db = queryset._db or DEFAULT_DB_ALIAS
+    if queryset.query.is_sliced and connections[db].features.supports_over_clause:
```
Comment (charettes) [inline django/db/models/fields/related_descriptors.py:89]: Any thoughts about error'ing out with a clearer error message than `AssertionError: Cannot filter a query once a slice has been taken.` when `not supports_over_clause` instead? Maybe `NotSupportedError("Prefetching from a limited queryset is only supported on backends that support window functions")`?

**Review by charettes — APPROVED** [review]
Thanks for the PR, any thoughts about raising a more explicit error message?

**Comment by felixxm** [comment]
@charettes Thanks :+1: I pushed edits.

File: django/db/models/fields/related_descriptors.py (hunk 1 above)
Comment (felixxm) [reply django/db/models/fields/related_descriptors.py]: Added

**Review by charettes — APPROVED** [review]
Awesome, thanks!

**Pull request merged**